Benchmarks/Benchmarks/NIOPosixBenchmarks/TCPEcho.swift

```diff
@@ -0,0 +1,95 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+@_spi(AsyncChannel) import NIOCore
+@_spi(AsyncChannel) import NIOPosix
+
+private final class EchoChannelHandler: ChannelInboundHandler {
+    fileprivate typealias InboundIn = ByteBuffer
+
+    func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+        context.writeAndFlush(data, promise: nil)
+    }
+}
+
+private final class EchoRequestChannelHandler: ChannelInboundHandler {
+    fileprivate typealias InboundIn = ByteBuffer
+
+    private let bufferSize = 10000
+    private let numberOfWrites: Int
+    private var batchCount = 0
+    private let data: NIOAny
+    private let readsCompletePromise: EventLoopPromise<Void>
+    private var receivedData = 0
+
+    init(numberOfWrites: Int, readsCompletePromise: EventLoopPromise<Void>) {
+        self.numberOfWrites = numberOfWrites
+        self.readsCompletePromise = readsCompletePromise
+        self.data = NIOAny(ByteBuffer(repeating: 0, count: self.bufferSize))
+    }
+
+    func channelActive(context: ChannelHandlerContext) {
+        for _ in 0..<self.numberOfWrites {
+            context.writeAndFlush(data, promise: nil)
+        }
+    }
+
+    func channelRead(context: ChannelHandlerContext, data: NIOAny) {
+        let buffer = self.unwrapInboundIn(data)
+        self.receivedData += buffer.readableBytes
+
+        if self.receivedData == self.numberOfWrites * self.bufferSize {
+            self.readsCompletePromise.succeed()
+            return
+        }
+    }
+}
+
+func runTCPEcho(numberOfWrites: Int) throws {
+    let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
```

Comment:
Should we just use the singleton?

Reply:
I wanted to constrain this to a single thread.

Reply:
I think the singleton does that although this is more explicit.

Reply:
The singleton is by default constrained to number of threads/cgroups AFAIK